docs-ref-conceptual/upcoming-breaking-changes.md

```diff
@@ -0,0 +1,21 @@
+---
+title: Upcoming Breaking Changes - Azure CLI | Microsoft Docs
+description: Learn about breaking changes coming to Azure CLI in the next breaking change release
+manager: mkluck
+author: dbradish-microsoft
+ms.author: dbradish
+ms.date: 09/27/2024
+ms.topic: article
+ms.service: azure-cli
+ms.tool: azure-cli
+ms.custom: devx-track-azurecli, seo-azure-cli
+keywords: azure cli updates, azure cli notes, azure cli versions, azure cli breaking changes
+---
+
+# Upcoming breaking changes in Azure CLI
+
+## dla
```

Comment:
What are our design options here?

Reply:
The document is generated from the Azure CLI repository and follows this structure:

```markdown
# Title

## Command Module or Extension

### `command group or command`

- List the breaking change items of the command.
```

The Breaking Change item displays the same as the warning in Azure CLI, following the pattern defined in breaking_change.py in azure-cli-core. You can find the relevant code here.
https://github.com/Azure/azure-cli/blob/3194af8f245f6d7e30ad70813c58eef779142482/src/azure-cli-core/azure/cli/core/breaking_change.py#L297-L508

Reply:
@ReaNAiveD, Looking at the *py you linked to, there are several breaking change classes which provide nice flexibility. However, which properties are **required**? I feel that we need to minimally supply the customer with a breaking change date or version, and remediation plan (i.e. "Use {this} instead").  I just went through conceptual content containing azCLI commands that will be deprecated, and the message in our autogen content does not contain a version or date, and only _sometimes_ provides a substitution command / remediation. I feel we need more information for a breaking change article. CCing @dcaro

Reply:
@dbradish-microsoft , Sorry for the confusion. These lines are the interface provided: https://github.com/Azure/azure-cli/blob/3194af8f245f6d7e30ad70813c58eef779142482/src/azure-cli-core/azure/cli/core/breaking_change.py#L548-L591
You can find how to use the interface in the doc: https://github.com/Azure/azure-cli/blob/dev/doc/how_to_introduce_breaking_changes.md#pre-announce-breaking-changes-1

For most types of breaking changes, the only required properties are the name of the command or arguments. By default, the breaking change `version` is the next breaking change window. I believe a version number is more suitable than a date because Azure CLI is a client tool rather than an online service. A version number is clearer for customers to understand when the breaking change will occur.

The remediation plan is an optional argument `redirect`, which is available for deprecation breaking changes. In some cases, the remediation plan might not be available until the breaking change is released, or not be provided at all. So I don't make it required.

I agree that more information is needed, but it depends on the respective service teams to provide it.